cmd/shcd.go

```diff
@@ -290,12 +314,109 @@ func (id Id) GenerateXname() (xn string) {
 
 		xn = x.String()
 
+	} else if strings.HasPrefix(id.CommonName, "ncn-") {
+
+		// Convert the rack to a string
+		cabString := id.Location.Rack
+
+		// Strip the "x"
+		_, cabNum := utf8.DecodeRuneInString(cabString)
+
+		// Convert to an int
+		cabinet, err := strconv.Atoi(cabString[cabNum:])
+
+		if err != nil {
+			log.Fatalln(err)
+		}
+
+		// Strip the u
```

Comment:
For sanitizing?

Not nit picking, but the comment doesn't really say _why_.

Reply:
The schema requires an `int` here, which you can't convert with the "u" there.